src/app/pages/project/project.component.ts

```diff
@@ -36,45 +39,100 @@ export class ProjectComponent {
-                this.initChartData(this.activeProject);
+                this.activeProject = this.projectList.length > 0 && this.projectService.projects.find((el) => el.id === this.activeProjectId)
+                this.activeProjectRepo = this.activeProject.projectRepos.find(repo => repo.isDefault) || this.activeProject.projectRepos[0];
+                this.initChartData();
             }
             // if page is accessed directly then fetch the data then filter the current report
             else {
                 this.projectService.getProjects().then((result) => {
                     this.projectList = result as Array<Project>;
-                    this.activeProject = this.projectList.length > 0 && this.projectService.projects.find((el) => el.id === activeReportId)
-                    this.initChartData(this.activeProject);
-                  })
+                    this.activeProject = this.projectList.length > 0 && this.projectService.projects.find((el) => el.id === this.activeProjectId)
+                    this.activeProjectRepo = this.activeProject.projectRepos.find(repo => repo.isDefault) || this.activeProject.projectRepos[0];
+                    this.initChartData();
+                })
             }
-
         });
     }
 
-    initChartData(activeProject) {
-        let defaultRepo = activeProject.projectRepos.find(repo => repo.isDefault) || activeProject.projectRepos[0];
-
+    initChartData() {
         //loop through all the metrics and assign them as data arrays
         //this.metrics = {Object.assign({}, activeProject.qualityReports[0].sonarQubeReport)};
         this.metrics = {};
-        for (var key in defaultRepo.qualityReports[0].sonarQubeReport) {
+        for (var key in this.activeProjectRepo.qualityReports[0].sonarQubeReport) {
             if(this.disabledMetrics.indexOf(key) === -1){
                 this.metrics[key] = [{name: key, series:[]}]
             }
-        }
-        defaultRepo.qualityReports.forEach((sonarReport) => {
-            const date = moment(sonarReport.updateDate).format('DD MM')
-            for (var _key in sonarReport.sonarQubeReport) {
-                if(this.disabledMetrics.indexOf(_key) === -1){
-                    this.metrics[_key][0].series.push({name: date, value: sonarReport.sonarQubeReport[_key]});
+        }           
+        
+        for(var _i = this.activeProjectRepo.qualityReports.length - 1; _i >= 0; _i--){
+                var sonarReport = this.activeProjectRepo.qualityReports[_i];
+                const date = moment(sonarReport.updateDate).format('DD MM')
+                for (var _key in sonarReport.sonarQubeReport) {
+                    if(this.disabledMetrics.indexOf(_key) === -1){
+                        this.metrics[_key][0].series.push({name: date, value: sonarReport.sonarQubeReport[_key]});
+                    }
                 }
-            }
-        })
+        }
+    
         console.log(this.metrics)
     }
+
+    onChange(newValue) {
```

Comment:
Let's us a more suggestive name, maybe **onActiveRepoChange(newActiveRepo)**? Also, remove the log bellow and the empty line.